x-pack/plugins/security_solution/public/common/utils/route/spy_routes.tsx

```diff
@@ -40,17 +41,24 @@ export const SpyRouteComponent = memo<
     useEffect(() => {
       if (pageName && !deepEqual(route.pathName, pathname)) {
         if (isInitializing && detailName == null) {
+          const routeAction = {
+            detailName,
+            flowTarget,
+            history,
+            pageName,
+            pathName: pathname,
+            state,
+            tabName,
+          };
           dispatch({
             type: 'updateRouteWithOutSearch',
-            route: {
-              detailName,
-              flowTarget,
-              history,
-              pageName,
-              pathName: pathname,
-              state,
-              tabName,
-            },
+            route:
```

Comment:
this is a nit but I'd do it this way:
```suggestion
            route: { 
              detailName,
              flowTarget,
              history,
              pageName,
              pathName: pathname,
              state,
              tabName,
              ...(pageName === SecurityPageName.administration ? { search: search ?? '' } : {})
          },
        }
```